modules/user_management.js

```diff
@@ -130,6 +130,52 @@ class UserManagement {
         }]
       }
     })
+
+    // Add a recital ID (role name) to the record of all participants
+    dispatcher.command('record', this._guild.id, async (authorMember, tokenized) => {
+      util.requireRole(authorMember, this._managementRole)
+
+      // p!record @role
+      const USAGE = `${this._config.get('commandPrefix') || 'p!'}record @role`
+      util.requireParameterCount(tokenized, 1, USAGE)
+      util.requireParameterFormat(tokenized[0], arg => arg.startsWith('<@&') && arg.endsWith('>'), USAGE)
+
+      const recitalRole = await this._guild.roles.fetch(tokenized[0].replace(/[<@&>]/g, ''))
+
+      // Check that input role is a recital role
+      // Recital roles must be below config.recitalUpperBound and above config.recitalLowerBound
+      const upperBound = await this._guild.roles.fetch(this._config.get('recitalUpperBoundId'))
+      const lowerBound = await this._guild.roles.fetch(this._config.get('recitalLowerBoundId'))
+      if ((upperBound !== undefined && recitalRole.comparePositionTo(upperBound) >= 0) ||
```

Comment:
!= null

Also, what happens if the config value is empty? Do we pass `undefined` to `fetch()`?

Reply:
If the config value is empty, the range would only be bounded on one side.

e.g. upperBound is @Performer, lowerBound is empty. Then the range of recital roles will be everything after @Performer to the very end of the list, including the very last role. 

This was done because I designed the bounds to be exclusive on both sides. This allows us to include either the very first role or very last role in the valid range, should it be necessary.